core/epochFlags.go

```diff
@@ -0,0 +1,232 @@
+	TransactionSignedWithTxHashFlag,
+	MetaProtectionFlag,
+	AheadOfTimeGasUsageFlag,
+	GasPriceModifierFlag,
+	RepairCallbackFlag,
+	ReturnDataToLastTransferFlagAfterEpoch,
+	SenderInOutTransferFlag,
+	StakeFlag,
+	StakingV2Flag,
+	StakingV2OwnerFlagInSpecificEpochOnly,
+	StakingV2FlagAfterEpoch,
+	DoubleKeyProtectionFlag,
+	ESDTFlag,
+	ESDTFlagInSpecificEpochOnly,
+	GovernanceFlag,
+	GovernanceFlagInSpecificEpochOnly,
+	DelegationManagerFlag,
+	DelegationSmartContractFlag,
+	DelegationSmartContractFlagInSpecificEpochOnly,
+	CorrectLastUnJailedFlagInSpecificEpochOnly,
+	CorrectLastUnJailedFlag,
+	RelayedTransactionsV2Flag,
+	UnBondTokensV2Flag,
+	SaveJailedAlwaysFlag,
+	ReDelegateBelowMinCheckFlag,
+	ValidatorToDelegationFlag,
+	IncrementSCRNonceInMultiTransferFlag,
+	ESDTMultiTransferFlag,
+	GlobalMintBurnFlag,
+	ESDTTransferRoleFlag,
+	BuiltInFunctionOnMetaFlag,
+	ComputeRewardCheckpointFlag,
+	SCRSizeInvariantCheckFlag,
+	BackwardCompSaveKeyValueFlag,
+	ESDTNFTCreateOnMultiShardFlag,
+	MetaESDTSetFlag,
+	AddTokensToDelegationFlag,
+	MultiESDTTransferFixOnCallBackFlag,
+	OptimizeGasUsedInCrossMiniBlocksFlag,
+	CorrectFirstQueuedFlag,
+	DeleteDelegatorAfterClaimRewardsFlag,
+	RemoveNonUpdatedStorageFlag,
+	OptimizeNFTStoreFlag,
+	CreateNFTThroughExecByCallerFlag,
+	StopDecreasingValidatorRatingWhenStuckFlag,
+	FrontRunningProtectionFlag,
+	PayableBySCFlag,
+	CleanUpInformativeSCRsFlag,
+	StorageAPICostOptimizationFlag,
+	ESDTRegisterAndSetAllRolesFlag,
+	ScheduledMiniBlocksFlag,
+	CorrectJailedNotUnStakedEmptyQueueFlag,
+	DoNotReturnOldBlockInBlockchainHookFlag,
+	AddFailedRelayedTxToInvalidMBsFlag,
+	SCRSizeInvariantOnBuiltInResultFlag,
+	CheckCorrectTokenIDForTransferRoleFlag,
+	FailExecutionOnEveryAPIErrorFlag,
+	MiniBlockPartialExecutionFlag,
+	ManagedCryptoAPIsFlag,
+	ESDTMetadataContinuousCleanupFlag,
+	DisableExecByCallerFlag,
+	RefactorContextFlag,
+	CheckFunctionArgumentFlag,
+	CheckExecuteOnReadOnlyFlag,
+	SetSenderInEeiOutputTransferFlag,
+	FixAsyncCallbackCheckFlag,
+	SaveToSystemAccountFlag,
+	CheckFrozenCollectionFlag,
+	SendAlwaysFlag,
+	ValueLengthCheckFlag,
+	CheckTransferFlag,
+	TransferToMetaFlag,
+	ESDTNFTImprovementV1Flag,
+	ChangeDelegationOwnerFlag,
+	RefactorPeersMiniBlocksFlag,
+	SCProcessorV2Flag,
+	FixAsyncCallBackArgsListFlag,
+	FixOldTokenLiquidityFlag,
+	RuntimeMemStoreLimitFlag,
+	RuntimeCodeSizeFixFlag,
+	MaxBlockchainHookCountersFlag,
+	WipeSingleNFTLiquidityDecreaseFlag,
+	AlwaysSaveTokenMetaDataFlag,
+	SetGuardianFlag,
+	RelayedNonceFixFlag,
+	ConsistentTokensValuesLengthCheckFlag,
+	KeepExecOrderOnCreatedSCRsFlag,
+	MultiClaimOnDelegationFlag,
+	ChangeUsernameFlag,
+	AutoBalanceDataTriesFlag,
+	FixDelegationChangeOwnerOnAccountFlag,
+	FixOOGReturnCodeFlag,
+	DeterministicSortOnValidatorsInfoFixFlag,
+}
+
+// CheckHandlerCompatibility checks if the provided handler is compatible with this mx-chain-core-go version
+func CheckHandlerCompatibility(handler EnableEpochsHandler) error {
```

Comment:
I talked a bit with @iulianpascalau and I think we can remove the flags from core, move it to mx-chain-go instead.
Also the Check can receive as parameter the list of flags to be checked.

Reply:
as discussed, will be fixed in the next PR